### PR TITLE
#856 Fix FintechConsentUiSmokeE2ETest class

### PR DESCRIPTION
**Pull request by ivaniuminov**
https://github.com/adorsys/open-banking-gateway/issues/856

### CODE REVIEW

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/WebDriverBasedUserInfoFintech.java [hunk 1]
```diff
@@ -82,6 +82,12 @@ public SELF user_max_musterman_in_consent_ui_sees_sca_select_and_confirm_type_em
         return self();
     }
 
+    public SELF user_in_consent_ui_sees_sca_select_and_confirm_type_email1_to_redirect_authorization(WebDriver driver) {
```
Comment (valb3r) [inline fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/WebDriverBasedUserInfoFintech.java:85]: Suggest to expand with
```
Index: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/WebDriverBasedUserInfoFintech.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/WebDriverBasedUserInfoFintech.java	(revision de38e7ad18d495e697ede1c5c108c6460410fef8)
+++ fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/WebDriverBasedUserInfoFintech.java	(date 1594386830000)
@@ -2,7 +2,11 @@
 import com.tngtech.jgiven.integration.spring.JGivenStage;
 import de.adorsys.opba.protocol.xs2a.tests.e2e.sandbox.servers.WebDriverBasedAccountInformation;
-import org.openqa.selenium.*;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.Select;
 import org.openqa.selenium.support.ui.WebDriverWait;
@@ -14,7 +18,9 @@
 import java.net.URI;
 import java.time.Duration;
-import static de.adorsys.fintech.tests.e2e.steps.FintechStagesUtils.*;
+import static de.adorsys.fintech.tests.e2e.steps.FintechStagesUtils.FINTECH_UI_URI;
+import static de.adorsys.fintech.tests.e2e.steps.FintechStagesUtils.OBA_USERNAME;
+import static de.adorsys.fintech.tests.e2e.steps.FintechStagesUtils.PIN;
 import static de.adorsys.opba.protocol.xs2a.tests.e2e.sandbox.servers.config.RetryableConfig.TEST_RETRY_OPS;
@@ -333,8 +339,11 @@
     }
     private void performClick(WebDriver driver, By identifier) {
-        wait(driver).until(ExpectedConditions.elementToBeClickable(identifier));
-        driver.findElement(identifier).click();
+        withRetry.execute(context -> {
+            wait(driver).until(ExpectedConditions.elementToBeClickable(identifier));
+            driver.findElement(identifier).click();
+            return null;
+        });
     }
     private void sendText(WebDriver driver, By identifier, String text) {
```
for stability

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java [hunk 2]
```diff
@@ -84,5 +114,102 @@ public SELF rest_assured_points_to_opba_server(String opbaServerUri) {
...
+    public SELF create_new_user_in_sandbox_tpp_management(String login, String password) {
+        String auth = login_into_tpp_management();
+        String userId = do_create_new_user_in_sandbox_tpp_management(login, password, auth);
+        create_account_for_user_in_sandbox_tpp_management(auth, userId);
+        String accountId = get_account_id_by_iban_in_sandbox_tpp_management(auth, iban);
+
+        return deposit_to_sandbox_tpp_management_user_account(auth, accountId);
+    }
+
+    private String login_into_tpp_management() {
```
Comment (valb3r) [inline fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:128]: Please follow
```
given()
    .somethingA()
.then()
    .somethingB()
```

formatting

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java [hunk 3]
```diff
@@ -84,5 +114,102 @@ public SELF rest_assured_points_to_opba_server(String opbaServerUri) {
...
+    }
+
+    @SneakyThrows
+    private String do_create_new_user_in_sandbox_tpp_management(String login, String password, String auth) {
+        String body = readResource("restrecord/tpp-ui-input/params/new-user-registration-body.json")
+                              .replace(USER_NAME_PLACEHOLDER, login)
+                              .replace(PASSWORD_PLACEHOLDER, password)
+                              .replace(EMAIL_PLACEHOLDER, login + "@example.com");
+
+        ExtractableResponse<Response> response = RestAssured
```
Comment (valb3r) [inline fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:149]: Please follow
```
given()
    .somethingA()
.then()
    .somethingB()
```

formatting

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java [hunk 4]
```diff
@@ -84,5 +114,102 @@ public SELF rest_assured_points_to_opba_server(String opbaServerUri) {
...
+
+        return response.body().jsonPath().getString("id");
+    }
+
+    private SELF create_account_for_user_in_sandbox_tpp_management(String auth, String userId) {
+        this.iban = Iban.random(CountryCode.DE).toString();
+        String body = readResource("restrecord/tpp-ui-input/params/new-user-new-account-registration.json")
+                              .replace(IBAN_PLACEHOLDER, iban);
+
+        RestAssured
```
Comment (valb3r) [inline fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:168]: Please follow
```
given()
    .somethingA()
.then()
    .somethingB()
```

formatting

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java [hunk 5]
```diff
@@ -84,5 +114,102 @@ public SELF rest_assured_points_to_opba_server(String opbaServerUri) {
...
+                .post(smokeConfig.getSandboxTppManagementServerUrl() + TPP_MANAGEMENT_CREATE_ACCOUNT_ENDPOINT)
+                .then()
+                .statusCode(HttpStatus.OK.value());
+
+        return self();
+    }
+
+    @SneakyThrows
+    private String get_account_id_by_iban_in_sandbox_tpp_management(String auth, String iban) {
+        ExtractableResponse<Response> response = RestAssured
```
Comment (valb3r) [inline fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:184]: Please follow
```
given()
    .somethingA()
.then()
    .somethingB()
```

formatting

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java [hunk 6]
```diff
@@ -84,5 +114,102 @@ public SELF rest_assured_points_to_opba_server(String opbaServerUri) {
...
+                                                         .extract();
+
+        this.accountResourceId = response.body().jsonPath().getString("id");
+        return accountResourceId;
+    }
+
+    private SELF deposit_to_sandbox_tpp_management_user_account(String auth, String accountId) {
+        String body = readResource("restrecord/tpp-ui-input/params/new-user-deposit-account.json");
+
+        RestAssured
```
Comment (valb3r) [inline fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:202]: Please follow
```
given()
    .somethingA()
.then()
    .somethingB()
```

formatting

Comment (ivaniuminov) [reply fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:202]: done

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java (hunk 5 above)
Comment (ivaniuminov) [reply fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:184]: done

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java (hunk 4 above)
Comment (ivaniuminov) [reply fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:168]: done

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java (hunk 3 above)
Comment (ivaniuminov) [reply fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:149]: done

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java (hunk 2 above)
Comment (ivaniuminov) [reply fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/FintechServer.java:128]: done

File: fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/WebDriverBasedUserInfoFintech.java (hunk 1 above)
Comment (ivaniuminov) [reply fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/WebDriverBasedUserInfoFintech.java:89]: Patch applied, thanks

**Pull request merged**